### PR TITLE
Added UserLocalStorageService

### PR DESCRIPTION
**Pull request by Stasiek99**

### CODE REVIEW

**Comment by kamazowe** [comment]
if I go to the site and the user has not been added yet, I get an error in the console

File: src/app/create-new-user/components/create-new-user/create-new-user.component.ts [hunk 1]
```diff
@@ -13,9 +14,29 @@ import {UserService} from "../../../user/user.service";
...
+  @ViewChild("form") form: any
   passedValues: any;
 
-  constructor(private dialog: MatDialog, private router: Router, private userService: UserService) {}
+  constructor(private dialog: MatDialog, private router: Router, private userService: UserService, private userLocalStorageService: UserLocalStorageService) {}
+
+  // ngOnInit(): void{
+  //   let localStorageUserData = window.localStorage["User"];
+  //   localStorageUserData = JSON.parse(localStorageUserData);
+  //
```
Comment (kamazowe) [inline src/app/create-new-user/components/create-new-user/create-new-user.component.ts:25]: comment ??

File: src/app/user/user-local-storage.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {Injectable} from "@angular/core";
+
+import {UserInterface} from "../user-presentation/interfaces/user-interface";
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class UserLocalStorageService {
+  createdUser: UserInterface | null = null;
```
Comment (kamazowe) [inline src/app/user/user-local-storage.service.ts:10]: this service should not contain application state

File: src/app/user/user-local-storage.service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  setUser(user: UserInterface) {
+    this.createdUser = user;
+    window.localStorage.setItem("User", JSON.stringify(user))
+  }
+
+  getUser(): UserInterface | null{
+    let localStorageFields = window.localStorage["User"];
```
[the remark this answers is not in the document]
Comment (kamazowe) [reply src/app/user/user-local-storage.service.ts:18]: return JSON.parse(localStorageFields);

File: src/app/user/user-local-storage.service.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {Injectable} from "@angular/core";
+
+import {UserInterface} from "../user-presentation/interfaces/user-interface";
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class UserLocalStorageService {
```
Comment (kamazowe) [inline src/app/user/user-local-storage.service.ts:9]: UserService should communicate with this service

File: src/app/user-presentation/components/user-presentation.component.ts [hunk 5]
```diff
@@ -10,11 +11,20 @@ import {UserService} from "../../user/user.service";
...
+    this.user = this.userLocalStorageService.getUser() as UserInterface;
     console.log(this.user)
+    if(this.user){
+      this.isButtonVisible = true;
+    }
+  }
+
+  deleteUser(){
+    window.localStorage.removeItem("User");
+    window.location.reload();
```
[the remark this answers is not in the document]
Comment (kamazowe) [reply src/app/user-presentation/components/user-presentation.component.ts:28]: if you can, then don't use the global object. try to do it another way

File: src/app/user-presentation/components/user-presentation.component.ts [hunk 6]
```diff
@@ -10,11 +11,20 @@ import {UserService} from "../../user/user.service";
...
-    this.user = this.userService.getUser() as UserInterface;
+    this.user = this.userLocalStorageService.getUser() as UserInterface;
     console.log(this.user)
+    if(this.user){
+      this.isButtonVisible = true;
+    }
+  }
+
+  deleteUser(){
+    window.localStorage.removeItem("User");
```
Comment (kamazowe) [inline src/app/user-presentation/components/user-presentation.component.ts:27]: why does the user presentation component know about localstorage? only UserLocalStorageService can know about localstorage.only in the above class you can use `window.localStorage`

File: src/app/user-presentation/components/user-presentation.component.html [hunk 7]
```diff
@@ -8,5 +8,8 @@
       <p>Name of your country: <b>{{user.country}}</b></p>
       <p>Your age: <b>{{user.age}}</b></p>
     </div>
+    <div fxLayout="row" fxLayoutGap="30px" fxLayout.sm="column" fxLayout.xs="column" fxLayoutAlign="center">
+      <button mat-raised-button *ngIf="this.isButtonVisible" (click)="deleteUser()">Delete User</button>
```
Comment (kamazowe) [inline src/app/user-presentation/components/user-presentation.component.html:12]: this.isButtonVisible > isButtonVisible

File: src/app/user-presentation/components/user-presentation.component.ts [hunk 8]
```diff
@@ -10,11 +11,20 @@ import {UserService} from "../../user/user.service";
...
   user: UserInterface | null = null;
+  isButtonVisible = false;
 
-  constructor(private userService: UserService) {}
+  constructor(private userService: UserService, private userLocalStorageService: UserLocalStorageService) {}
 
   ngOnInit(){
-    this.user = this.userService.getUser() as UserInterface;
+    this.user = this.userLocalStorageService.getUser() as UserInterface;
     console.log(this.user)
```
Comment (kamazowe) [inline src/app/user-presentation/components/user-presentation.component.ts:20]: cls?

File: src/app/shared/shared-form/user-form.component.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-shared-form',
+  templateUrl: './user-form.component.html',
+  styleUrls: ['./user-form.component.scss']
+})
+export class UserFormComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (kamazowe) [inline src/app/shared/shared-form/user-form.component.ts:12]: lint error

**Review by kamazowe — CHANGES_REQUESTED** [review]

File: src/app/user/user-state.service.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import {Injectable} from "@angular/core";
+
+import {UserInterface} from "../user-presentation/interfaces/user-interface";
+import {UserLocalStorageService} from "./user-local-storage.service";
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class UserStateService {
```
Comment (kamazowe) [inline src/app/user/user-state.service.ts:10]: good job, this service looks cool

File: src/app/user/user-state.service.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  constructor(private userLocalStorageService: UserLocalStorageService) {}
```
Comment (kamazowe) [inline src/app/user/user-state.service.ts:11]: I propose to initially assign the value from localstorage to the createdUser field in constructor
.then you can simplify the `getUser` method

File: src/app/user/user-state.service.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  getUser(): UserInterface | null{
+    if(this.userLocalStorageService.getUser()){
+      return this.userLocalStorageService.getUser();
+    }
+    else {
+      return this.createdUser;
+    }
+  }
+
+  deleteUser(){
```
Comment (kamazowe) [inline src/app/user/user-state.service.ts:28]: Should we only delete data from localstorage?

**Review by kamazowe — CHANGES_REQUESTED** [review]

File: src/app/create-new-user/components/create-new-user/create-new-user.component.ts [hunk 13]
```diff
@@ -15,7 +15,7 @@ export class CreateNewUserComponent {
   @ViewChild("f") userDataForm!: NgForm;
   passedValues: any;
 
-  constructor(private dialog: MatDialog, private router: Router, private userService: UserService) {}
+  constructor(private dialog: MatDialog, private router: Router, private userService: UserStateService) {}
 
   openDialog(): void {
    let dialogRef = this.dialog.open(DialogCreateUserComponent, {
```
[the remark this answers is not in the document]
Comment (Stasiek99) [reply src/app/create-new-user/components/create-new-user/create-new-user.component.ts:21]: When I remove 22nd line of code, 30th line does not execute

[the remark this answers is not in the document]
Comment (Stasiek99) [reply src/app/create-new-user/components/create-new-user/create-new-user.component.ts:21]: When I delete 22nd line of code, 30th does not execute

**Review by kamazowe — APPROVED** [review]

**Pull request merged**